teps/0052-tekton-results-automated-run-resource-cleanup.md

```diff
@@ -46,17 +46,27 @@ keep track of.
 
 Today, most users view Tekton Run history via the Tekton Pipeline API (e.g.
 kubectl, tkn, dashboard, etc). However since all CRD objects are
-[stored in etcd](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/),
-this creates a problem for the Pipeline controller - as the build history grows
-over time, so does the corpus of data the controller needs to handle in its data
-plane. Over time this grows unbounded, significantly degrading the performance
-of the cluster as more Runs are completed.
+[stored in the Kubernetes API](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/),
+the data plane that is used for managing Runs by the controller is also the
+source of data users use for historical data.
+
+Over time this grows unbounded as more Runs are completed. For Tekton, this is
+particularly aggravated due to the automated nature of Runs being created (i.e.
+triggered builds).
 
 Most of this data is unnecessary for the Pipeline controller to hold on to -
-once a Run is complete it no longer needs to be stored in etcd since the
-Pipeline controller will take no further action on it. With Tekton Results, we
-now have a place to store long term results independent of the Pipeline
-controller, even if the data is removed from the etcd data plane.
+
+- Once a Run is complete it no longer needs to be stored in the Kubernetes API
+  since the Pipeline controller will take no further action on it. Keeping this
+  data around is particularly painful on a controller restart, since the
+  controller will attempt to re-reconcile every object, even though all we would
+  do is check if they are completed and move on.
+- Storing large amounts of CRD objects leads to noticeable latency in Kubernetes
+  API requests made by users (i.e. kubectl) - See [Controller List Latency] for
```

Comment:
i think this is intended to be a link but it no workie

Reply:
Fixed!